### ISSUE TITLE
About negative-aware

### ISSUE THREAD

**Issue by huangxu1991** [NVlabs/DiffusionNFT#24]
Nice work! especially have an interest in the idea of negaive-aware. I'm wondering whether the idea can be applied to simplified offline fine-tuning.

Suppose a pretrained model: pi, and collect N positive examples which we want. In addition, we collect extra N negative examples which should be avoided when model generating.

<img width="1041" height="554" alt="Image" src="https://github.com/user-attachments/assets/4bc7d115-b240-4860-a4ce-dca3143d3906" />

so the fine-tuning flow-matching model (rectified version) process:
x_t = (1 - t) * x0 + t * x1 (x0 -> clean, x1 -> noise)
v_t = x1 - x0

train dataset:
    {N pos, N neg}

initialize:
    V_theta = pi  # params updated by BP
    V_old = pi  # params updated through EMA

```
for each update iteration:
    1. random sample a batch B

    2. for example in positive part -> B_pos:
            x0, r = example  # x0 is clean source data. reward r equals to 1, meaning positve / good data, which are wanted
            sample noise x1, draw timestep t between (0, 1)
            x_t = (1 - t) * x0 + t * x1
            implicit positive velocity: v_pos = (1 - beta) * V_old(x_t, t, condition) + beta * V_theta(x_t, t, condition)
            params update: theta = theta - lambda * \grad {square(v_pos - v_t)}

    3. for example in negative part -> B_neg:
            x0, r = example  # x0 is clean source data. reward r equals to 0, meaning negative / bad data, which should be avoided
            sample noise x1, draw timestep t between (0, 1)
            x_t = (1 - t) * x0 + t * x1
            implicit negaive velocity: v_neg = (1 + beta) * V_old(x_t, t, condition) - beta * V_theta(x_t, t, condition)
            params update: theta = theta - lambda * \grad {square(v_neg - v_t)}

    4. update V_old ...  # EMA style
```

**Comment by zhengkw18** [comment]
That might work, but cannot be explained as online RL. In our setting we have clear signals of reward which could be observed during training. You can try and see if using offline data works in your setting.

**Comment by huangxu1991** [comment]
I'll try it ASAP
